xcape/cape_fortran.py

```diff
@@ -1,10 +1,10 @@
 
-def cape(p_2d, t_2d, td_2d, p_s, t_s, td_s, pres_lev_pos, source, ml_depth, adiabat, pinc, type_grid):
+def cape(p_2d, t_2d, td_2d, p_s, t_s, td_s, flag_1d, pres_lev_pos, source, ml_depth, adiabat, pinc, type_grid):
     
                 
     # nlev has to be the first dimension
     # nlev here is the number of levels in 3d variables (without surface level)
-    nlev, ngrid = p_2d.shape
+    nlev, ngrid = t_2d.shape
```

Comment:
I changed all shape definition to be a function of t_2d or temperature more in general, so that p can take different shapes.